tornado_jsonapi/resource.py

```diff
@@ -315,33 +364,49 @@ def update(self, id_, attributes):
         with (yield self.cursor(self.connection, transaction=True)) as cursor:
             cur = dbapiext.execute_f(
                 cursor,
-                'update %s set %X where id = %X',
-                self._tablename, attributes, id_)
+                "update %s set %X where id = %X",
+                self._tablename,
+                attributes,
+                id_,
+            )
             if is_future(cur):
                 cur = yield cur
         with (yield self.cursor(self.connection)) as cursor:
-            cur = dbapiext.execute_f(cursor, 'select %s from %s where id = %X',
-                                     self.columns + ['id'],
-                                     self._tablename, id_)
+            cur = dbapiext.execute_f(
+                cursor,
+                "select %s from %s where id = %X",
+                self.columns + ["id"],
+                self._tablename,
+                id_,
+            )
             if is_future(cur):
                 cur = yield cur
             return DBAPI2Resource.ResourceObject(self, cur.fetchone())
 
     @gen.coroutine
     def delete(self, id_):
         with (yield self.cursor(self.connection, transaction=True)) as cursor:
-            cur = dbapiext.execute_f(cursor, 'delete from %s where id = %X',
-                                     self._tablename, id_)
+            cur = dbapiext.execute_f(
+                cursor, "delete from %s where id = %X", self._tablename, id_
+            )
             if is_future(cur):
                 cur = yield cur
             return cur.rowcount
 
     @gen.coroutine
-    def list_(self):
+    def list_(self, limit=0, page=0):
         with (yield self.cursor(self.connection)) as cursor:
-            cur = dbapiext.execute_f(cursor, 'select %s from %s',
-                                     self.columns + ['id'], self._tablename)
+            cur = dbapiext.execute_f(
+                cursor, "select %s from %s", self.columns + ["id"], self._tablename
+            )
             if is_future(cur):
                 cur = yield cur
             rows = cur.fetchall()
             return [DBAPI2Resource.ResourceObject(self, row) for row in rows]
+
+    @gen.coroutine
+    def list_count_(self):
+        count = self.list_()
+        while is_future(count):
+            count = yield count
+        return len(count)
```

Comment:
This method looks quite ineffective. Perhaps it would be better to issue `select count(1)` SQL statement.

Reply:
Let me know if I implemented this correctly